fabtests/functional/cq_data.c

```diff
@@ -43,6 +43,7 @@ static int run_test()
 	size_t size = 1000;
 	struct fi_cq_data_entry comp = {0};
 	struct fi_rma_iov remote;
+	uint64_t mask = (0x1ULL << (fi->domain_attr->cq_data_size * 8)) - 1;
```

Comment:
If cq_data_size is 8, this left shifts 1 by 64, which is undefined.

Reply:
For your consideration:

```c
size_t cq_data_bits = fi->domain_attr->cq_data_size * 8;
uint64_t mask = (((uint64_t) (cq_data_bits < 64)) << (cq_data_bits & 63)) - 1;
```

Reply:
Thanks both. Please see the update.